src/solution_transfer/meshfree_solution_transfer.C

```diff
@@ -37,68 +37,6 @@ namespace libMesh
 template <typename T>
 class DenseVector;
 
-// Helper function for doing the projection
-class MeshlessInterpolationFunction : public FunctionBase<Number>
-{
-public:
-  MeshlessInterpolationFunction (const MeshfreeInterpolation & mfi,
-                                 Threads::spin_mutex & mutex) :
-    _mfi(mfi),
-    _mutex(mutex)
-  {}
-
-  /**
-   * The move/copy ctor and destructor are defaulted for this class.
-   */
-  MeshlessInterpolationFunction (MeshlessInterpolationFunction &&) = default;
-  MeshlessInterpolationFunction (const MeshlessInterpolationFunction &) = default;
-  virtual ~MeshlessInterpolationFunction () = default;
-
-  /**
-   * This class contains const references so it can't be assigned.
-   */
-  MeshlessInterpolationFunction & operator= (const MeshlessInterpolationFunction &) = delete;
-  MeshlessInterpolationFunction & operator= (MeshlessInterpolationFunction &&) = delete;
```

Comment:
We should probably add the above 5 move/copy/assignment methods to the new consolidated class.  If @jwpeterson was enough of a stickler to put them in this internal-library-use-only version, I can't imagine him letting us get away with leaving them out of the exposed version.